tenable/io/v3/vm/scanners.py

```diff
@@ -0,0 +1,364 @@
+        """
+        return self._api.get(
+            "scanners/{}/scans".format(self._check("id", id, int))
+        ).json()["scans"]
+
+    def search(self):
+        """
+        Search endpoint introduced in v3.
+
+        :devportal:`scanners: search <scanners-search>`
+
+        Returns:
+            :obj:`list`:
+                Iterator Class object
+                TODO Implementation of base iterator class UWSearchIterator needs to be updated at v3/base/iterator
+        Examples:
+            TODO
+        """
+        raise NotImplementedError
+    
+    def list(self):
+        '''
+        Retrieves the list of scanners.
+
+        :devportal:`scanners: list <scanners-list>`
+
+        Returns:
+            :obj:`list`:
+                List of scanner resource records.
+
+        Examples:
+            >>> for scanner in tio.scanners.list():
+            ...     pprint(scanner)
+        '''
+        return self._api.get('scanners').json()['scanners']
+
+    def toggle_link_state(self, id, linked):
+        """
+        Toggles the scanner's activated state.
+
+        :devportal:`scanners: toggle-link-state <scanners-toggle-link-state>`
+
+        Args:
+            id (int): The unique identifier for the scanner
+            linked (bool):
+                The link status of the scanner.  Setting to `False` will disable
+                the link, whereas setting to `True` will enable the link.
+
+        Returns:
+            :obj:`None`:
+                The status change was successful.
+
+        Examples:
+            to deactivate a linked scanner:
+
+            >>> tio.v3.vm.scanners.toggle_link_state(1, False)
+        """
+        self._api.put(
+            "scanners/{}/link".format(self._check("id", id, int)),
+            json={"link": int(self._check("linked", linked, bool))},
+        )
+
+    def get_permissions(self, id):
+        """
+        Returns the permission list for a given scanner.
+
+        Args:
+            id (int): The unique identifier for the scanner.
+
+        Returns:
+            :obj:`dict`:
+                The permissions resource for the scanner
+
+        Examples:
+            >>> tio.v3.vm.scanners.get_permissions(1)
+        """
+        return self._api.permissions.list("scanner", self._check("id", id, int))
+
+    def edit_permissions(self, id, *acls):
+        """
+        Modifies the permissions list for the given scanner.
+
+        Args:
+            id (int):The unique identifier for the scanner.
+            *acls (dict): The permissions record(s) for the scanner.
+
+        Returns:
+            :obj:`None`:
+                The permissions have been updated successfully.
+
+        Examples:
+            >>> tio.v3.vm.scanners.edit_permissions(1,
+            ...     {'type': 'default, 'permissions': 16},
+            ...     {'type': 'user', 'id': 2, 'permissions': 16})
+        """
+        self._api.permissions.change("scanner", self._check("id", id, int), *acls)
```

Comment:
I'd assume that this should be using `self._api.v3.vm.permissions.change`

_check usage should be deprecated

Reply:
Updated the method with the following
`return NotImplementedError( "This method will be updated once Permissions API's are implemented in v3" )`

Yes it will be` self._api.v3.vm.permissions.list` once the code of Permissions API for pyTenable is implemented in v3

removed _check and added typing